02week/ticTacToe.js

```diff
@@ -24,39 +24,80 @@ function printBoard() {
 }
 
 function horizontalWin() {
-  // Your code here
+  return board[0].every(square => square === playerTurn) ||
+         board[1].every(square => square === playerTurn) ||
+         board[2].every(square => square === playerTurn);
 }
 
 function verticalWin() {
-  // Your code here
+  return [board[0][0], board[1][0], board[2][0]].every(square => square === playerTurn) ||
+         [board[0][1], board[1][1], board[2][1]].every(square => square === playerTurn) ||
+         [board[0][2], board[1][2], board[2][2]].every(square => square === playerTurn);
 }
 
 function diagonalWin() {
-  // Your code here
+  return [board[0][0], board[1][1], board[2][2]].every(square => square === playerTurn) ||
+         [board[0][2], board[1][1], board[2][0]].every(square => square === playerTurn);
 }
 
 function checkForWin() {
-  // Your code here
+  if (horizontalWin()) {
+    printBoard();
+    return true;
+  } else if (verticalWin()) {
+    printBoard();
+    return true;
+  } else if (diagonalWin()) {
+    printBoard();
+    return true;
+  }
+  return false;
 }
 
 function ticTacToe(row, column) {
-  // Your code here
+  const validValue = (myIndex) => {
+    const values = [0,1,2];
+    return values.some(validIndex => myIndex == validIndex);
+  }
+
+  if (validValue(row) && validValue(column)) {
```

Comment:
I am not sure how to get this to work. I have tried several different ways and I keep getting errors in node 
any advice?